src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/ClientWebSocketOptions.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Do we need to specialize this file for the browser?  Or instead could ClientWebSocket.ConnectAsync just throw PNSE if it finds options set that aren't supported?

Reply:
The list of dependencies from the desktop version is much bigger (size savings) but it'd be nice to extract the common code to the shared file

Reply:
> The list of dependencies from the desktop version is much bigger (size savings)

Only if the consumer uses them, though, right?  Otherwise we should be able to ensure the relevant stuff is linked away?

Reply:
Yes if we don't call them internally, for example, to check if there were set.